lib/sgov/sgov-adapter.ts

```diff
@@ -0,0 +1,227 @@
+
+  protected resourceGroupCache = new Map<string, string[] | undefined>();
+
+  constructor(endpoint: string, httpFetch: HttpFetch) {
+    this.endpoint = endpoint;
+    this.httpFetch = httpFetch;
+  }
+
+  setIriProvider(iriProvider: IriProvider): void {
+    this.iriProvider = iriProvider;
+  }
+
+  async search(searchString: string): Promise<PimClass[]> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const source = new SparqlQueryRdfSource(this.httpFetch, this.endpoint, searchQuery(searchString));
+    await source.query();
+
+    const results = await source.property("__search_results", "__has_search_result");
+
+    let sorted = [];
+    for (const result of results) {
+      const resultWrap = RdfSourceWrap.forIri(result.value, source);
+      await this.cacheResourceGroup(resultWrap);
+      sorted.push({
+        sort: Number((await resultWrap.property("__order"))[0].value),
+        cls: await loadSgovClass(resultWrap, this.iriProvider),
+      });
+    }
+    sorted = sorted.sort((a, b) => a.sort - b.sort).map(p => p.cls);
+
+    if (IRI_REGEXP.test(searchString)) {
+      const classByIri = await this.getClass(searchString);
+      if (classByIri) {
+        sorted = [classByIri, ...sorted];
+      }
+    }
+
+    return sorted;
+  }
+
+  async getClass(cimIri: string): Promise<PimClass | null> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const source = new SparqlQueryRdfSource(this.httpFetch, this.endpoint, getClassQuery(cimIri));
+    await source.query();
+    const entity = RdfSourceWrap.forIri(cimIri, source);
+
+    if (!await isSgovClass(entity)) {
+      return null;
+    }
+    await this.cacheResourceGroup(RdfSourceWrap.forIri(cimIri, source));
+    return await loadSgovClass(entity, this.iriProvider);
+  }
+
+  async getSurroundings(cimIri: string): Promise<ReadOnlyMemoryStore> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const sources = getSurroundingsQueries(cimIri).map(query => new SparqlQueryRdfSource(this.httpFetch, this.endpoint, query));
+    await Promise.all(sources.map(q => q.query()));
+    const source = FederatedSource.createExhaustive(sources);
+    const resources = await this.loadPimEntitiesGraphFromEntity(cimIri, source);
+    return new ReadOnlyMemoryStore(resources);
+  }
+
+  public async getFullHierarchy(cimIri: string): Promise<ReadOnlyMemoryStore> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const source = new SparqlQueryRdfSource(this.httpFetch, this.endpoint, getHierarchyQuery(cimIri));
+    await source.query();
+    const resources = await this.loadPimEntitiesGraphFromEntity(cimIri, source);
+    return new ReadOnlyMemoryStore(resources);
+  }
+
+  /**
+   * Returns a glossary info for the resource.
+   * @param cimIri
+   */
+  public async getResourceGroup(cimIri: string): Promise<string[]> {
```

Comment:
Why is this method ```public``` why others are not? I believe ```public``` is the default modifier, so it is not necessary to write it here.